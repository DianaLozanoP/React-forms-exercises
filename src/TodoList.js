import { useState } from "react";
import { v4 as uuid } from "uuid"
import ToDo from "./ToDo";
import NewTodoForm from "./NewToDoForm";

const TodoList = () => {
    const INITIAL_STATE = [
        { activity: "Laundry", id: uuid() }]
    const [toDos, setToDos] = useState(INITIAL_STATE)
    const addToDo = ({ activity }) => {
        setToDos(toDos => [...toDos, { activity, id: uuid() }])
    }
    const removeToDo = id => {
        const newToDos = toDos.filter(todo => todo.id !== id)
        setToDos(newToDos)
    }
    return (
        <div>
            <h3>To Do List</h3>
            <NewTodoForm addToDo={addToDo} />
            <div>
                <ul>
                    <li>
                        {toDos.map(toDo => <ToDo key={toDo.id} id={toDo.id} activity={toDo.activity} removeToDo={removeToDo} />)}
                    </li>
                </ul>
            </div>
        </div>
    )
}

export default TodoList;