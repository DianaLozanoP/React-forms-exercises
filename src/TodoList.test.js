import { fireEvent, render } from '@testing-library/react'
import TodoList from "./TodoList"

test("it renders without crashing", () => {
    render(<TodoList />)
})

test("matches snapshot", () => {
    const { asFragment } = render(<TodoList />)
    expect(asFragment()).toMatchSnapshot();
})

test("should add a new activity to the list", () => {
    const { getByLabelText, container, getByText } = render(<TodoList />)
    const input = getByLabelText("Enter a new to do")
    const btn = getByText("Add it")
    const ul = container.querySelector('ul')
    fireEvent.change(input, { target: { value: "Fill taxes" } })
    fireEvent.click(btn)
    expect(ul).toHaveTextContent('Fill taxes')
})