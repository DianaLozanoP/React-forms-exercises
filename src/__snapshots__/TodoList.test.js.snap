// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      To Do List
    </h3>
    <form>
      <label
        for="activity"
      >
        Enter a new to do
      </label>
      <input
        id="activity"
        name="activity"
        placeholder="Replace lightbulb"
        type="text"
        value=""
      />
      <button>
        Add it
      </button>
    </form>
    <div>
      <ul>
        <li>
          <div>
            Laundry
            <button
              style="position: relative; background-color: lavender;"
            >
              X
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
